pkg/apis/apps/types.go

```diff
@@ -195,6 +195,27 @@ type StatefulSetStatus struct {
 	// newest ControllerRevision.
 	// +optional
 	CollisionCount *int32
+
+	// Represents the latest available observations of a statefulset's current state.
+	Conditions []StatefulSetCondition
+}
+
+type StatefulSetConditionType string
+
+// StatefulSetCondition describes the state of a statefulset at a certain point.
+type StatefulSetCondition struct {
+	// Type of statefulset condition.
+	Type StatefulSetConditionType
+	// Status of the condition, one of True, False, Unknown.
+	Status api.ConditionStatus
+	// The last time this condition was updated.
+	LastUpdateTime metav1.Time
```

Comment:
Do we want to remove LastUpdateTime now?  It would mean this is inconsistent with others, but at least we wouldn't propagate it.

Reply:
Is there consensus on this?

@kow3ns @janetkuo thoughts?

Reply:
`LastUpdateTime` was added to DeploymentCondition to replace `LastHeartbeatTime` or `lastProbeTime` we see in other conditions (ref https://github.com/kubernetes/kubernetes/pull/19343#issuecomment-255880158 and https://github.com/kubernetes/kubernetes/pull/35529#issuecomment-256098418). It's useful for  users to understand what is going on. 

We haven't added it to ReplicaSetCondition, though.

Reply:
Based on our discussion - looks like we can remove it from here, since this type isn't in use yet. There's no strong reason to add it only to remove it later.